src/Microsoft.DocAsCode.Build.TableOfContents/BuildTocDocumentStepBase.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.TableOfContents
+{
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+
+    using Microsoft.DocAsCode.Build.Common;
+    using Microsoft.DocAsCode.Common;
+    using Microsoft.DocAsCode.DataContracts.Common;
+    using Microsoft.DocAsCode.Plugins;
+
+    public abstract class BuildTocDocumentStepBase : BaseDocumentBuildStep
+    {
+        #region IDocumentBuildStep
+
+        public override void Build(FileModel model, IHostService host)
+        {
+            var toc = (TocItemViewModel)model.Content;
+            TocRestructureUtility.Restructure(toc, host.TableOfContentRestructions);
+            BuildCore(toc, model, host);
+            ReportBuildDependency(model, host);
+            model.Content = toc;
+            // todo : metadata.
+        }
+
+        #endregion
+
+        #region Abstract methods
+
+        public abstract void ReportBuildDependency(FileModel model, IHostService host);
```

Comment:
or "BuildPostActions"? Any suggestions?

---
In reply to: [121313850](https://github.com/dotnet/docfx/pull/1750#discussion_r121313850) [](ancestors = 121313850)